src/utils/uniswap.js

```diff
@@ -24,28 +24,23 @@ import {
 } from '@uniswap/sdk';
 import { toChecksumAddress } from '@netgum/utils';
 import { BigNumber } from 'bignumber.js';
+import { getEnv } from 'configs/envConfig';
 
 import ROUTER_ABI from 'abi/uniswapRouter.json';
-import { isProdEnv } from 'utils/environment';
 import { encodeContractMethod } from 'services/assets';
 import { ETH } from 'constants/assetsConstants';
 import { ALLOWED_SLIPPAGE } from 'constants/exchangeConstants';
 
 import type { Asset } from 'models/Asset';
 
-const isMainnet = isProdEnv;
-export const chainId = isMainnet ? ChainId.MAINNET : ChainId.KOVAN;
+type UniswapChainId = ChainId.MAINNET | ChainId.KOVAN;
+
+export const getChainId = (): UniswapChainId => {
+  return getEnv().NETWORK_PROVIDER === 'homestead' ? ChainId.MAINNET : ChainId.KOVAN;
```

Comment:
```suggestion
  return isProdEnv() ? ChainId.MAINNET : ChainId.KOVAN;
```
This check seems to be based on implementation detail. My suggestion would be to encapsulate the prod/staging env check into a app-wide function (either replacing isProdEnv or a new function) located around `configs/envConfig.js`.

Reply:
I didn't want to change `isProdEnv` initially because it relied on build type, not env vars loaded. I thought there might have been some reason to that, but it seems it's safe to just check envConfig, not the build type. Changed to a function.